servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/WriteStreamSubscriber.java

```diff
@@ -366,6 +366,8 @@ void sourceTerminated(@Nullable Throwable cause) {
             }
             this.failureCause = cause;
             setFlag(SOURCE_TERMINATED);
+            // Mark the subscription as CANCELLED to prevent propagating cancel from channelClosed
+            subscriptionUpdater.getAndSet(WriteStreamSubscriber.this, CANCELLED);
```

Comment:
No, all these changes to the internal state of `AllPromise` are happening on a single event loop thread.

Reply:
Also, `channelOutboundClosed` and `channelClosed` are always sequential because they are on the same event loop thread. `channelOutboundClosed` comes from `userEventTriggered`, `channelClosed` comes from `channelInactive`.